File: src/main/java/org/shark/example/domain/user/application/RegisterUserService.java
package org.shark.example.domain.user.application;

import lombok.RequiredArgsConstructor;
import org.shark.example.domain.user.model.UserDto;
import org.shark.example.domain.user.model.UserRepository;
import org.springframework.stereotype.Service;

import java.util.UUID;

@Service
@RequiredArgsConstructor
public class RegisterUserService {

    private final UserRepository userRepository;

    public void register(String name, String email) {
        if (userRepository.existsByEmail(email)) {
            throw new IllegalArgumentException("Email already used.");
        }
        UserDto user = new UserDto(UUID.randomUUID().toString(), name, email);
        userRepository.save(user);
    }
}

File: src/main/kotlin/org/shark/example/domain/user/application/RegisterUserService.kt
package org.shark.example.domain.user.application

import org.shark.example.domain.user.model.UserDto
import org.shark.example.domain.user.model.UserRepository
import org.springframework.stereotype.Service
import java.util.UUID

@Service
class RegisterUserService(
    private val userRepository: UserRepository
) {
    fun register(name: String, email: String) {
        if (userRepository.existsByEmail(email)) {
            throw IllegalArgumentException("Email already used.")
        }
        val user = UserDto(UUID.randomUUID().toString(), name, email)
        userRepository.save(user)
    }
}
